tola_management/programadmin.py

```diff
@@ -417,21 +413,115 @@ class ProgramAdminViewSet(viewsets.ModelViewSet):
     pagination_class = Paginator
     permission_classes = [permissions.IsAuthenticated, HasProgramAdminAccess]
 
+    @classmethod
+    def base_queryset(cls):
+        """this adds annotations for the "users" and "organizations" links using annotations
+        
+            For performance reasons, looking up every user with permission to see the program individually for
+            all 20+ in a paginated set was costly.  This annotates the same information in one spendy query
+        """
+        # start with the correctly annotated-for-rf queryset:
+        queryset = Program.rf_aware_objects.all()
+        queryset = queryset.annotate(
+            # this counts users who are not mercy corps users (partner access) who have been assigned:
+            program_access_users_count=models.functions.Coalesce( # coalesce so None is 0 (summable later)
+                models.Subquery(
+                    ProgramAccess.objects.filter(
+                        program=models.OuterRef('pk'),
+                        tolauser__organization_id__gt=1
```

Comment:
Would it make sense to move away from using the hard coded id number of MC's org for these kinds of queries? 
 I assume we've done this all over the app (I'm sure I'm guilty of similar things), but setting up constants somewhere to store this info might be good.  Totally understand if you don't want to do it as part of this patch though.

Reply:
this is a really good thought, I've been using id != 1 or id = 1 for a while, but that's terrible practice.  I'm gonna file a backstage ticket to look for other instances, but I fixed it in this one